src/Chess.php

```diff
@@ -31,12 +31,18 @@ class Chess
     /** @var array<string, string> */
     protected $sanMoveCache = [];
 
-    public function __construct(?string $fen = Board::DEFAULT_POSITION)
+    /**
+     * @param array<int, History>|null $history
+     */
+    public function __construct(?string $fen = Board::DEFAULT_POSITION, ?array $history = null)
     {
         $this->board = new Board();
         if (null !== $error = $this->load($fen ?? Board::DEFAULT_POSITION)) {
             throw new \InvalidArgumentException(\sprintf('Invalid fen: %s. Error: %s', $fen, $error));
         }
+        if (null !== $history) {
+            $this->history = $history;
```

Comment:
maybe we can validate input with something like this:
```php
$this->history = array_map(fn(History $history): History => $history, $history);
```

Reply:
We can't use fn, unless we raise PHP requirements